pkg/grpc/service/studyflow_endpoints.go

```diff
@@ -290,17 +285,14 @@ func (s *studyServiceServer) GetAssignedSurveys(ctx context.Context, req *api_ty
 				cs.ProfileId = profileID
 				resp.Surveys = append(resp.Surveys, cs)
 
-				sDef := types.Survey{}
-				for _, def := range studySurveys {
-					if def.Current.SurveyDefinition.Key == cs.SurveyKey {
-						sDef = def
-						break
-					}
+				sDef, err := s.studyDBservice.FindCurentSurveyDef(req.InstanceId, study.Key, cs.SurveyKey, true)
```

Comment:
What about a cache logic inside the service itself ?

Reply:
I was thinking of adding a map data object (map of map) for study / survey key, at least for current user - so if already fetched in the context of the current call, it could be reused

Reply:
Added a simple caching mechanism for the current call, to avoid obviously redundant DB calls.